src/common/analyticsTrackingConstants.js

```diff
@@ -0,0 +1,92 @@
+/**
```

Comment:
yes to the first comment. This file is meant to construct declarative event names that are consistently and semantically formatted. 

> mouse event scope produced by the scope function
```js
MOUSE:{
  CLICK: "MOUSE__CLICK",
 HOVER: "MOUSE__HOVER"
 scope: "MOUSE"
}
```

The scope function simply allows for a shorthand instead of writing out the entire event name and also makes it more declarative in component implementation. 

```js 
const MyComponent = ({tracking:{EVENT_CONSTANTS: { MOUSE }) => { ... }
export default withAnalyticsTracking(MyComponent);
```

By looking at the component above you can easily tell that it will fire off mouse events then look up the mouse events schemas in `event_schemas/MOUSE/...` for more detail

Reply:
Couldn't you just define these constants up front instead of piling a bunch of functions to compute it?
```
const MOUSE = {
  MOUSE:{
  CLICK: "MOUSE__CLICK",
  HOVER: "MOUSE__HOVER"
}
```
Having the `scope` property is a little confusing because we'll presumably be pushing the scope stack with each event, too.

For the second point, why do we need to pass events to components if we have them all defined as constants? If the purpose is to be able to tell what events a component fires, maybe we could just do it with a docstring.

Reply:
I thought about using a docstring but the json schema is alot to put into a docstring

Reply:
> ... we'll presumably be pushing the scope stack with each event, too

What do you mean by "scope stack" ?

Reply:
You don't need to put the whole schema definiton into the docstring, just the event name.

Reply:
The scope stack like `['app', 'dashboard', 'save button']` that gets passed down through event properties.